### PR TITLE
koord-manager: move expiring metrics into util, add prom metrics

### PR DESCRIPTION
**Pull request by saintube**
…for slo-controller

### Ⅰ. Describe what this PR does

1. utils: move expiring metrics structs from pkg koordlet into pkg util.
2. koord-manager: add common prom metrics for slo-controller.

### V. Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `make test`

### CODE REVIEW

File: pkg/slo-controller/metrics/node_resource.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+	"github.com/koordinator-sh/koordinator/pkg/util/metrics"
+)
+
+func init() {
+	MustRegister(NodeResourceCollectors...)
+}
+
+var (
+	NodeResourceRunPluginStatus = metrics.NewGCCounterVec("node_resource_run_plugin_status", prometheus.NewCounterVec(prometheus.CounterOpts{
```
Comment (zwzhang0107) [inline pkg/slo-controller/metrics/node_resource.go:31]: only metrics related to node/pod object should use GC metrics (has NodeKey/PodKey).

Comment (saintube) [reply pkg/slo-controller/metrics/node_resource.go]: fixed

File: pkg/slo-controller/metrics/node_resource.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+var (
+	NodeResourceRunPluginStatus = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Subsystem: SLOControllerSubsystem,
+		Name:      "node_resource_run_plugin_status",
+		Help:      "the status count of running node resource plugins",
+	}, []string{PluginKey, StatusKey, ReasonKey})
+
+	NodeExtendedResourceAllocatableInternal = metrics.NewGCGaugeVec("node_extended_resource_allocatable_internal", prometheus.NewGaugeVec(prometheus.GaugeOpts{
```
Comment (zwzhang0107) [inline pkg/slo-controller/metrics/node_resource.go:37]: naming, e.g. xxx_internal

**Review by hormes — APPROVED** [review]

**Pull request merged**